### DISCUSSION TITLE
Pass environment variables from step description into a shell step

### DISCUSSION THREAD

**Discussion by vadimkantorov** [Vampire/setup-wsl#4] [Q&A] [answered]
Is it possible to pass the env variables from step description into shell "wsl-bash" step? I tried and it failed, but maybe I did something sloppy.

Another question: is it possible to specify working directory as a property of the shell step? Two common choices are: case-insensitive mount such as `/mnt/c/Users/user/` (currently the default I think), and the case-sensitive (`/home/user/`). Explicitly choosing between the two would be good (or somehow providing an env variable for a simple cd into the one or the other)

Example of my workflow: https://github.com/vadimkantorov/perlwslv1buildtest/blob/master/.github/workflows/build.yml

**Comment by Vampire** [comment]
Two details about the example you showed.

1. To install the build-essentials package, you can also use the `additional-packages` input of my action if you want.

2. The default user is `root`, so you don't need to use `sudo`.

Regarding environment variables, the environment variables you set, are set in Windows. Almost none of the Windows environment variables are forwarded to WSL. You need to explicitly specify in a separate environment variable which environment variables to forward to WSL and how to convert them (like paths or path lists) like shown for example at https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/.

Regarding working directory, you can define the working directory of a run step as input, I think that should also be the working directory inside WSL then, but I'm not fully sure about that. I'm not sure what you mean with the case sensitive and insensitive part and to choose what. Can you elaborate on that, maybe with an example?

**Comment by Vampire** [comment, marked as answer]
Regarding your questions here also a little demo workflow run: https://github.com/Vampire/setup-wsl/runs/1469130942?check_suite_focus=true

Regarding the directories, I verified that you are in the exact directory within WSL where you are outside, so setting the `working-directory` input works as expected.

Maybe you misinterpret that Windows environment and WSL environment are two completely separate things.
The user home directory on Windows is the home directory of the Windows user, which is usually found at `c:\Users\<user>\` and thus mounted in WSL at `/mnt/c/Users/<user>/`.
The directory `/home/foo` in WSL is usually the home directory of the user `foo` wihin the Linux distribution which is completely separate from an eventually existing user `foo` in Windows.
If you want to switch to the home directory of the user within WSL, you can simply do `cd` or if you want it a bit more verbose `cd ~`.
If you want to switch to the Windows home directory of the Windows user that invoked the WSL command, you can do so as I have shown in the demo workflow run by setting up the `USERPROFILE` env variable to be forwarded to WSL and then either use `wslpath` to convert it to the WSL path or right away configure the forwarding to already translate the path. (I showed both in that example run).

**Comment by vadimkantorov** [comment]
Thank you for your help! I improved my workflow with `additional-packages` and `WSLENV: USERPROFILE`. This solves these both problems. Thanks for maintainting this action!
